Sources/SwifterSwift/Foundation/LocaleExtensions.swift

```diff
@@ -17,6 +17,15 @@ public extension Locale {
         return Locale(identifier: "en_US_POSIX")
     }
 
+    /// SwifterSwift: Returns bool value indicating if locale has 12h format.
+    var is12HourTimeFormat: Bool {
+        let dateFormatter = DateFormatter()
+        dateFormatter.timeStyle = .short
+        dateFormatter.dateStyle = .none
+        dateFormatter.locale = self
+        let dateString = dateFormatter.string(from: Date())
+        return dateString.contains(dateFormatter.amSymbol) || dateString.contains(dateFormatter.pmSymbol)
```

Comment:
FYI, this will fail if the user removes the AM/PM symbols in the Language & Region => Advanced system preferences.

The only reliable check is:

```swift
extension Locale {
	var uses12HourTimeFormat: Bool {
		DateFormatter.dateFormat(fromTemplate: "j", options: 0, locale: self)?.contains("a") ?? false
	}
}
```

Reply:
@sindresorhus interesting 🤔
I'm going to reference this under its issue: [#773](https://github.com/SwifterSwift/SwifterSwift/issues/773). I'd be grateful if you can open a PR with the fix!

cc @DimaZava

Reply:
@omaralbeik @sindresorhus 
<img width="2560" alt="Screenshot 2020-01-21 at 1 03 14  (2)" src="https://user-images.githubusercontent.com/1781631/72795313-7a962f80-3c4e-11ea-910d-f9ded2322db7.png">